pkgs/development/python-modules/pooch/default.nix

```
@@ -0,0 +1,52 @@
+{ stdenv
+, buildPythonPackage
+, isPy27
+, fetchPypi
+, pytest
+, packaging
+, appdirs
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "pooch";
+  version = "1.0.0";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1k2vinlhkzl7lzhvbz20x3a2r2zqqila0yxg3a3fax2r6qxbxxzi";
+  };
+
+  propagatedBuildInputs = [ packaging appdirs requests ];
+
+  checkInputs = [ pytest ];
+  checkPhase = ''
+    pytest -v -k "not pooch_custom_url \
+      and not pooch_download \
+      and not pooch_logging_level \
+      and not pooch_update \
+      and not pooch_corrupted \
+      and not check_availability \
+      and not check_availability_on_ftp \
+      and not downloader \
+      and not ftp_downloader \
+      and not test_fetch \
+      and not decompress \
+      and not decompress_fails \
+      and not extractprocessor_fails \
+      and not processors \
+      and not processors \
+      and not processor_multiplefiles \
+      and not processor_multiplefiles \
+      and not integration"
```

pytest will match if it's part of a substring, you can clean this up a little
```suggestion
    pytest -v -k "not pooch_custom_url \
      and not pooch_download \
      and not pooch_logging_level \
      and not pooch_update \
      and not pooch_corrupted \
      and not check_availability \
      and not downloader \
      and not test_fetch \
      and not decompress \
      and not extractprocessor_fails \
      and not processor \
      and not integration"
```

---

also, pytest6 will not support new lines, please use `pytestCheckHook` + `disabledTests` ... which I need up date the docs to show an example

```
  checkInputs = [ pytestCheckHook ];
  disabledTests = [
    "pooch_custom_url"
    ...
  ];
```

---

`checkInputs = [ pytestCheckHook ];`

gives me

```
undefined variable 'pytestCheckHook'
```

---

sorry, take a look at https://github.com/NixOS/nixpkgs/blob/233dc9c7d1893efb04c974e5d91f3b4a04fad4fe/doc/languages-frameworks/python.section.md#using-pytestcheckhook

but it also needs to be added to the top of the expression as if it was another package

---

example expression https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/xdg/default.nix

---

If you prefer watching videos, I made one on packaging python packages https://www.youtube.com/watch?v=jXd-hkP4xnU

---

thanks for the pointers :)